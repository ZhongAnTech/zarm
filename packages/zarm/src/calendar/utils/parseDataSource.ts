function parseDataSource(months, locale) {
  const dataSource = {};
  for (let i = 0; i < months.length; i++) {
    const year = months[i].getFullYear();
    const month = months[i].getMonth();
    if (!dataSource[year]) {
      dataSource[year] = {
        value: year,
        label: year,
        children: [
          {
            value: month,
            label: locale?.months[month],
          },
        ],
      };
    } else {
      dataSource[year].children.push({
        value: month,
        label: locale?.months[month],
      });
    }
  }
  return dataSource;
}

export default parseDataSource;
