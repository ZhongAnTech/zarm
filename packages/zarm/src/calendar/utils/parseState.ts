import DateTool from '../../utils/date';

const parseState = (props: {
  min?: Date;
  max?: Date;
  value?: Date | Date[];
  defaultValue?: Date | Date[];
  multiple: boolean;
}) => {
  const { defaultValue, multiple } = props;
  let { value } = props;

  let tmpValue!: Date[];

  value = value || defaultValue;
  value = (Object.prototype.toString.call(value) === '[object Array]'
    ? value
    : (value && [value]) || []) as Date[];

  // 注掉该逻辑，强制根据 multiple 控制节点个数，后面改进
  // tmpValue = value.map(item => DateTool.parseDay(item));
  tmpValue = value.slice(0, multiple ? 2 : 1).map((item: Date) => DateTool.parseDay(item));
  // 排序过滤
  tmpValue = tmpValue!.sort((item1: Date, item2: Date) => +item1 - +item2);
  const min = props.min ? DateTool.parseDay(props.min) : new Date();
  const startMonth = DateTool.cloneDate(min, 'dd', 1);
  const max = props.max ? DateTool.parseDay(props.max) : DateTool.cloneDate(min, 'y', 1);
  const endMonth = DateTool.cloneDate(max, 'dd', DateTool.getDaysByDate(max));

  // min、max 排序
  const duration = [min, max].sort((item1: Date, item2: Date) => +item1 - +item2);

  const tmp = {
    value: tmpValue,
    min: duration[0],
    max: duration[1],
    startMonth,
    endMonth,
    // 是否是入参更新(主要是月份跨度更新，需要重新定位)
    refresh: false,
    // 注掉该逻辑，强制根据 multiple 控制节点个数，后面改进
    // steps:Math.max(tmp.value.length, tmp.defaultValue.length);
    steps: multiple ? 2 : 1,
    // 初始化点击步数
    multiple,
  };

  return tmp;
};

export default parseState;
