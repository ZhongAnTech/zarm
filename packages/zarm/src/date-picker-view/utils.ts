import dayjs from 'dayjs';
import isLeapYear from 'dayjs/plugin/isLeapYear';
import isoWeek from 'dayjs/plugin/isoWeek';
import isoWeeksInYear from 'dayjs/plugin/isoWeeksInYear';
import isFunction from 'lodash/isFunction';
import * as React from 'react';
import { ConfigContext } from '../config-provider';
import { PickerColumn, PickerColumnItem } from '../picker-view';
import { ColumnType, COLUMN_TYPE, DatePickerFilter, MERIDIEM, RenderLabel } from './interface';

dayjs.extend(isoWeek);
dayjs.extend(isoWeeksInYear);
dayjs.extend(isLeapYear);

export const generateMeridiemHour = (hour: number, hasMeridiem: boolean) => {
  if (!hasMeridiem) return hour;
  if (hour === 0) hour = 12;
  if (hour > 12) hour -= 12;
  return hour;
};

export const useRenderLabel = (renderLabel?: RenderLabel): RenderLabel => {
  const { locale: globalLocale } = React.useContext(ConfigContext);
  const locale = globalLocale?.DatePickerView;

  return React.useCallback(
    (type, value) => {
      if (isFunction(renderLabel)) {
        return renderLabel(type, value);
      }

      switch (type) {
        case COLUMN_TYPE.YEAR:
          return value + locale.year;
        case COLUMN_TYPE.MONTH:
          return value + locale.month;
        case COLUMN_TYPE.WEEK:
          return value + locale.week;
        case COLUMN_TYPE.WEEK_DAY:
          return locale.weeks[value - 1];
        case COLUMN_TYPE.DAY:
          return value + locale.day;
        case COLUMN_TYPE.HOUR:
          return value + locale.hour;
        case COLUMN_TYPE.MINUTE:
          return value + locale.minute;
        case COLUMN_TYPE.SECOND:
          return value + locale.second;
        case COLUMN_TYPE.MERIDIEM:
          return value === MERIDIEM.AM ? locale.am : locale.pm;
        default:
          return value.toString();
      }
    },
    [renderLabel],
  );
};

export const padZero = (value: string | number, length = 2) => {
  return String(value).padStart(length, '0');
};

export const generateDatePickerColumns = (
  selectedValue: Date,
  min: Date,
  max: Date,
  columnType: ColumnType[],
  renderLabel: RenderLabel,
  filter?: DatePickerFilter,
) => {
  const hasMeridiem = columnType.includes(COLUMN_TYPE.MERIDIEM);

  const date = dayjs(selectedValue);
  const minDate = dayjs(min);
  const maxDate = dayjs(max);

  const minYear = min.getFullYear();
  const minMonth = min.getMonth() + 1;
  const minDay = min.getDate();
  const minWeek = minDate.isoWeek();
  const minWeekday = minDate.isoWeekday();
  const minHour = min.getHours();
  const minMinute = min.getMinutes();
  const minSecond = min.getSeconds();

  const maxYear = max.getFullYear();
  const maxMonth = max.getMonth() + 1;
  const maxDay = max.getDate();
  const maxWeek = maxDate.isoWeek();
  const maxWeekday = maxDate.isoWeekday();
  const maxHour = max.getHours();
  const maxMinute = max.getMinutes();
  const maxSecond = max.getSeconds();

  const selectedYear = date.year();
  const selectedMonth = date.month() + 1;
  const selectedDay = date.date();
  const selectedWeek = date.isoWeek();
  const selectedHour = date.hour();
  const selectedMinute = date.minute();
  const selectedYearWeeks = dayjs(`${selectedYear}/01/01`).isoWeeksInYear();

  const isInMinYear = selectedYear === minYear;
  const isInMaxYear = selectedYear === maxYear;
  const isInMinMonth = isInMinYear && selectedMonth === minMonth;
  const isInMaxMonth = isInMaxYear && selectedMonth === maxMonth;
  const isInMinDay = isInMinMonth && selectedDay === minDay;
  const isInMaxDay = isInMaxMonth && selectedDay === maxDay;
  const isInMinWeek = isInMinYear && selectedWeek === minWeek;
  const isInMaxWeek = isInMaxYear && selectedWeek === maxWeek;
  const isInMinHour = isInMinDay && selectedHour === minHour;
  const isInMaxHour = isInMaxDay && selectedHour === maxHour;
  const isInMinMinute = isInMinHour && selectedMinute === minMinute;
  const isInMaxMinute = isInMaxHour && selectedMinute === maxMinute;

  const firstDayInSelectedMonth = dayjs(date).startOf('month');

  const generateColumns = <T extends PickerColumnItem>(
    from: number,
    to: number,
    type: ColumnType,
    iteratee: (index: number) => T,
  ) => {
    let options: T[] = [];
    for (let i = from; i <= to; i++) {
      options.push(iteratee(i));
    }

    if (isFunction(filter)) {
      options = options.filter((option) =>
        filter(type, {
          get value() {
            return option.value as number;
          },
          get date() {
            return selectedValue;
          },
        }),
      );
    }

    return options;
  };

  const generateMeridiemHourColumn = (hour: number) => {
    if (!hasMeridiem) return hour;
    if (hour > 12) hour -= 12;
    return hour;
  };

  const columns: PickerColumn = [];

  columnType?.forEach((type) => {
    switch (type) {
      case COLUMN_TYPE.YEAR:
        columns.push(
          generateColumns(minYear, maxYear, COLUMN_TYPE.YEAR, (index) => ({
            value: index,
            label: renderLabel(COLUMN_TYPE.YEAR, index),
          })),
        );
        break;
      case COLUMN_TYPE.MONTH:
        columns.push(
          generateColumns(
            isInMinYear ? minMonth : 1,
            isInMaxYear ? maxMonth : 12,
            COLUMN_TYPE.MONTH,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.MONTH, index),
            }),
          ),
        );
        break;
      case COLUMN_TYPE.WEEK:
        columns.push(
          generateColumns(
            isInMinYear ? minWeek : 1,
            isInMaxYear ? maxWeek : selectedYearWeeks,
            COLUMN_TYPE.WEEK,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.WEEK, index),
            }),
          ),
        );
        break;
      case COLUMN_TYPE.WEEK_DAY:
        columns.push(
          generateColumns(
            isInMinWeek ? minWeekday : 1,
            isInMaxWeek ? maxWeekday : 7,
            COLUMN_TYPE.WEEK_DAY,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.WEEK_DAY, index),
            }),
          ),
        );
        break;
      case COLUMN_TYPE.DAY:
        columns.push(
          generateColumns(
            isInMinMonth ? minDay : 1,
            isInMaxMonth ? maxDay : firstDayInSelectedMonth.daysInMonth(),
            COLUMN_TYPE.DAY,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.DAY, index),
            }),
          ),
        );
        break;
      case COLUMN_TYPE.MERIDIEM:
        columns.push(
          generateColumns(MERIDIEM.AM, MERIDIEM.PM, COLUMN_TYPE.MERIDIEM, (index) => ({
            value: index,
            label: renderLabel(COLUMN_TYPE.MERIDIEM, index),
          })),
        );
        break;
      case COLUMN_TYPE.HOUR:
        columns.push(
          generateColumns(
            generateMeridiemHourColumn(isInMinDay ? minHour : 0),
            generateMeridiemHourColumn(isInMaxDay ? maxHour : 23),
            COLUMN_TYPE.HOUR,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.HOUR, generateMeridiemHour(index, hasMeridiem)),
            }),
          ),
        );
        break;
      case COLUMN_TYPE.MINUTE:
        columns.push(
          generateColumns(
            isInMinHour ? minMinute : 0,
            isInMaxHour ? maxMinute : 59,
            COLUMN_TYPE.MINUTE,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.MINUTE, index),
            }),
          ),
        );
        break;
      case COLUMN_TYPE.SECOND:
        columns.push(
          generateColumns(
            isInMinMinute ? minSecond : 0,
            isInMaxMinute ? maxSecond : 59,
            COLUMN_TYPE.SECOND,
            (index) => ({
              value: index,
              label: renderLabel(COLUMN_TYPE.SECOND, index),
            }),
          ),
        );
        break;
    }
  });

  return columns;
};

export const dateToNumberArray = (value: Date, columnType: ColumnType[]) => {
  const date = dayjs(value);
  const hasWeek = columnType?.includes(COLUMN_TYPE.WEEK);
  const hasMeridiem = columnType?.includes(COLUMN_TYPE.MERIDIEM);
  return columnType?.map((type) => {
    switch (type) {
      case COLUMN_TYPE.YEAR:
        return hasWeek ? date.isoWeekYear() : date.year();
      case COLUMN_TYPE.MONTH:
        return date.month() + 1;
      case COLUMN_TYPE.DAY:
        return date.date();
      case COLUMN_TYPE.WEEK:
        return date.isoWeek();
      case COLUMN_TYPE.WEEK_DAY:
        return date.isoWeekday();
      case COLUMN_TYPE.MERIDIEM:
        return date.hour() >= 12 ? MERIDIEM.PM : MERIDIEM.AM;
      case COLUMN_TYPE.HOUR:
        return generateMeridiemHour(date.hour(), hasMeridiem);
      case COLUMN_TYPE.MINUTE:
        return date.minute();
      case COLUMN_TYPE.SECOND:
        return date.second();
      default:
        return undefined;
    }
  });
};

export const numberArrayToDate = (
  currentValue: Date,
  changedPickerValue: number[],
  columnType: ColumnType[],
  level: number,
) => {
  const date = dayjs(currentValue);
  const types = Object.fromEntries(
    columnType.map((type, index) => [type, changedPickerValue[index]]),
  );
  const currentColumnType = columnType[level];

  const getMeridiemHour = (meridiem: number, hour: number) => {
    if (meridiem === MERIDIEM.AM && hour >= 12) {
      hour -= 12;
    }
    if (meridiem === MERIDIEM.PM && hour < 12) {
      hour += 12;
    }
    return hour;
  };

  const getHour = (hour: number) => {
    const meridiemIndex = columnType?.findIndex((i) => i === COLUMN_TYPE.MERIDIEM);
    const meridiemValue = changedPickerValue[meridiemIndex] as number;
    return getMeridiemHour(meridiemValue, hour);
  };

  const year = types.year ?? date.isoWeekYear() ?? 1900;
  const month = types.month ?? date.month() + 1 ?? 0;
  const week = types.week ?? 1;
  const weekDay = types['week-day'] ?? 1;
  const day = types.day ?? date.date() ?? 1;
  const hour = types.hour ?? 0;
  const minute = types.minute ?? 0;
  const second = types.second ?? 0;

  const selectedMonth = month - 1;
  const selectedDay = Math.min(day, dayjs(new Date(year, selectedMonth, 1)).daysInMonth());
  const selectedHour = getHour(hour);
  const selectedDate = dayjs().hour(selectedHour).minute(minute).second(second);

  if (columnType.includes(COLUMN_TYPE.WEEK)) {
    if (currentColumnType === COLUMN_TYPE.MONTH) {
      return selectedDate.year(year).month(selectedMonth).toDate();
    }
    if (currentColumnType === COLUMN_TYPE.DAY) {
      return selectedDate.year(year).isoWeek(week).date(selectedDay).toDate();
    }
    return selectedDate.year(year).isoWeek(week).isoWeekday(weekDay).toDate();
  }

  return selectedDate.year(year).month(selectedMonth).date(selectedDay).toDate();
};
