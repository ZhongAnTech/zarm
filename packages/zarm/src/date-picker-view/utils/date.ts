export const getGregorianCalendar = (year, month, day, hour, minutes, seconds) => {
  return new Date(year, month, day, hour, minutes, seconds);
};

export const cloneDate = (date) => {
  return new Date(+date);
};

// 获取当月天数
export function getDaysInMonth(date) {
  return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
}

export const setMonth = (date, month) => {
  date.setDate(Math.min(date.getDate(), getDaysInMonth(new Date(date.getFullYear(), month))));
  date.setMonth(month);
};

// 补齐格式
export const pad = (n) => {
  return n < 10 ? `0${n}` : n;
};
