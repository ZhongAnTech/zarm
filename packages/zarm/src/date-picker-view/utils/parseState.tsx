import type { BaseDatePickerViewProps, DatePickerValue } from '../PropsType';

const isExtendDate = (date?: DatePickerValue): Date | '' => {
  if (date instanceof Date) {
    return date;
  }

  if (!date) {
    return '';
  }

  const isTime = /^\d{2}:\d{2}$/.test(date);
  if (isTime) {
    return new Date(`${new Date().getFullYear()} ${date}`);
  }

  return new Date(date.toString().replace(/-/g, '/'));
};

const parseState = (props: BaseDatePickerViewProps) => {
  const date = props.value && isExtendDate(props.value);
  const defaultDate = props.defaultValue && isExtendDate(props.defaultValue);
  const wheelDefault = props.wheelDefaultValue && isExtendDate(props.wheelDefaultValue);

  return {
    date: date || defaultDate,
    wheelDefault,
  };
};

export { isExtendDate, parseState };
