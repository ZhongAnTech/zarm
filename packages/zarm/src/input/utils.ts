export function getValue(props, defaultValue: string | number = ''): string | number {
  if (typeof props.value !== 'undefined' && props.value !== null) {
    return props.value;
  }
  if (typeof props.defaultValue !== 'undefined' && props.defaultValue !== null) {
    return props.defaultValue;
  }
  return defaultValue;
}
