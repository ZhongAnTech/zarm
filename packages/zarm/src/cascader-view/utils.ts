import { FieldNames } from '../picker-view/interface';
import { resolvedFieldNames } from '../picker-view/utils';
import type {
  BaseCascaderViewProps,
  CascaderItem,
  CascaderOption,
  CascaderValue,
} from './interface';

export const parseState = (props: BaseCascaderViewProps) => {
  const { value, defaultValue } = props;
  const currentValue: CascaderValue[] = [];

  return {
    value: value || defaultValue,
    currentValue,
  };
};

export const parseItems = (
  options: CascaderOption[],
  value: CascaderValue[],
  fieldNames?: Partial<FieldNames>,
) => {
  const field = resolvedFieldNames(fieldNames);
  const items: CascaderItem[] = [];
  const traverse = (memo: CascaderOption[], depth = 0) => {
    memo.forEach((option) => {
      const children = option[field.children] as CascaderOption[] | undefined;
      const currentValue = value[depth];
      if (option[field.value] === currentValue) {
        items[depth] = option;

        if (Array.isArray(children) && children.length > 0) {
          traverse(children, depth + 1);
        }
      }
    });
  };

  traverse(options);

  return items;
};
