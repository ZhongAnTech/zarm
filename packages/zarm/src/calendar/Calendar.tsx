/* eslint-disable operator-linebreak */
import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseCalendarProps } from './PropsType';
import CalendarMonthView from './Month';
import parseState from './utils/parseState';
import DateTool from '../utils/date';

export interface CalendarProps extends BaseCalendarProps {
  prefixCls?: string;
  className?: string;
}

export interface CalendarStates {
  value: Date[];
  min: Date;
  max: Date;
  startMonth: Date;
  endMonth: Date;
  // 是否是入参更新(主要是月份跨度更新，需要重新定位)
  refresh: boolean;
  // 注掉该逻辑，强制根据 multiple 控制节点个数，后面改进
  // steps:Math.max(tmp.value.length; tmp.defaultValue.length);
  // steps 是总的选择的个数 via zouhuan
  steps: number;
  // 初始化点击步数
  // step 是为了扩展的，以后如果是三选，四选之类的，用这个，step 标注每次事件是第几次选择 via zouhuan
  step: number;
  multiple: boolean;
}

export default class CalendarView extends PureComponent<CalendarProps, CalendarStates> {
  static displayName = 'CalendarView';

  static defaultProps: CalendarProps = {
    prefixCls: 'za-calendar',
    multiple: false,
    min: new Date(),
    dateRender: (date: Date) => date.getDate(),
    disabledDate: () => false,
  };

  // 当前月份dom数据缓存
  private nodes?: object;

  constructor(props: CalendarProps) {
    super(props);
    this.nodes = {};
  }

  state = {
    ...parseState(this.props),
    step: 1,
  };

  componentDidMount() {
    this.anchor();
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (
      ('value' in nextProps && nextProps.value !== prevState.prevValue) ||
      ('multiple' in nextProps && nextProps.multiple !== prevState.prevMultiple) ||
      ('min' in nextProps && nextProps.min !== prevState.prevMin) ||
      ('max' in nextProps && nextProps.max !== prevState.prevMax)
    ) {
      return {
        ...parseState(nextProps),
        step: prevState.step ? 1 : prevState.step,
        refresh: !prevState.refresh,
        prevValue: nextProps.value,
        prevMax: nextProps.max,
        prevMin: nextProps.min,
        prevMultiple: nextProps.multiple,
      };
    }
    return null;
  }

  componentDidUpdate(_prevProps: CalendarProps, prevState: CalendarStates) {
    const { refresh } = this.state;
    if (refresh !== prevState.refresh) {
      this.anchor();
    }
  }

  // 日期点击事件，注意排序
  handleDateClick = (date: Date) => {
    const { step, steps, value } = this.state;
    const { onChange } = this.props;
    if (step === 1) {
      value.splice(0, value.length);
    }
    value[step - 1] = date;
    value.sort((item1: Date, item2: Date) => +item1 - +item2);
    this.setState(
      {
        value,
        step: step >= steps ? 1 : step + 1,
      },
      () => {
        step >= steps && typeof onChange === 'function' && onChange(value);
      },
    );
  };

  // 月历定位
  anchor = () => {
    const { value } = this.state;
    const target = value[0] || new Date();
    const key = `${target.getFullYear()}-${target.getMonth()}`;
    const node = this.nodes![key];
    if (node && Object.prototype.toString.call(node.anchor) === '[object Function]') {
      node.anchor();
    }
  };

  // 生成星期条
  renderWeekBar = () => {
    const { prefixCls, locale } = this.props;
    const content = locale!.weeks.map((week) => (
      <li key={week} className={`${prefixCls}__bar__item`}>
        {week}
      </li>
    ));
    return <ul className={`${prefixCls}__bar`}>{content}</ul>;
  };

  renderMonth = (dateMonth: Date) => {
    const { value, min, max } = this.state;
    const { prefixCls, dateRender, disabledDate } = this.props;
    const key = `${dateMonth.getFullYear()}-${dateMonth.getMonth()}`;
    return (
      <CalendarMonthView
        prefixCls={prefixCls}
        key={key}
        min={min}
        max={max}
        value={value}
        dateMonth={dateMonth}
        dateRender={dateRender}
        disabledDate={disabledDate}
        onDateClick={this.handleDateClick}
        ref={(n) => {
          this.nodes![key] = n;
        }}
      />
    );
  };

  // 生成日历内容
  renderMonths() {
    const { prefixCls } = this.props;
    const { startMonth, max } = this.state;
    const arr = Array.from({ length: DateTool.getMonthCount(startMonth, max) });
    const content = arr.map((_item, i) => this.renderMonth(DateTool.cloneDate(startMonth, 'm', i)));
    return <section className={`${prefixCls}__body`}>{content}</section>;
  }

  render() {
    const { prefixCls, className } = this.props;
    return (
      <div className={classnames(prefixCls, className)}>
        {this.renderWeekBar()}
        {this.renderMonths()}
      </div>
    );
  }
}
