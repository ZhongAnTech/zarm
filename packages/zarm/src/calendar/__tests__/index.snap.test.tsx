import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { mocked } from 'ts-jest/utils';
import Calendar from '../index';
import { date7, date2, date6 } from '../../../tests/testData/date';
import parseState from '../utils/parseState';
import DateTool from '../../utils/date';
import CalendarMonthView from '../Month';

jest.mock('../utils/parseState');
const mockedParseState = mocked(parseState);

describe('Calendar snapshot', () => {
  afterAll(() => {
    jest.resetAllMocks();
    jest.restoreAllMocks();
  });
  it('calendar render', () => {
    jest.spyOn(CalendarMonthView.prototype, 'checkStatus').mockReturnValue({
      disabled: false,
      isSelected: true,
      isRange: true,
      rangeStart: true,
      rangeEnd: true,
    });
    mockedParseState.mockReturnValue({
      value: [date2, date6],
      min: date2,
      max: date6,
      startMonth: date2,
      endMonth: date6,
      refresh: false,
      steps: 1,
      multiple: false,
    });
    jest.spyOn(DateTool, 'cloneDate').mockReturnValue(date7);
    jest.spyOn(DateTool, 'getMonthCount').mockReturnValue(1);

    const wrapper = mount(
      <Calendar defaultValue={date2} min={date2} max={date6} multiple={false} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
