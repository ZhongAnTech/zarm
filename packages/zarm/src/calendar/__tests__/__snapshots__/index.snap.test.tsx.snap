// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar snapshot calendar render 1`] = `
<CalendarView
  dateRender={[Function]}
  defaultValue={2017-11-03T07:00:00.000Z}
  disabledDate={[Function]}
  max={2017-11-06T07:00:00.000Z}
  min={2017-11-03T07:00:00.000Z}
  multiple={false}
  prefixCls="za-calendar"
>
  <div
    className="za-calendar"
  >
    <ul
      className="za-calendar__bar"
    >
      <li
        className="za-calendar__bar__item"
        key="日"
      >
        日
      </li>
      <li
        className="za-calendar__bar__item"
        key="一"
      >
        一
      </li>
      <li
        className="za-calendar__bar__item"
        key="二"
      >
        二
      </li>
      <li
        className="za-calendar__bar__item"
        key="三"
      >
        三
      </li>
      <li
        className="za-calendar__bar__item"
        key="四"
      >
        四
      </li>
      <li
        className="za-calendar__bar__item"
        key="五"
      >
        五
      </li>
      <li
        className="za-calendar__bar__item"
        key="六"
      >
        六
      </li>
    </ul>
    <section
      className="za-calendar__body"
    >
      <CalendarMonthView
        dateMonth={2018-05-06T00:00:00.000Z}
        dateRender={[Function]}
        disabledDate={[Function]}
        key="2018-4"
        max={2017-11-06T07:00:00.000Z}
        min={2017-11-03T07:00:00.000Z}
        onDateClick={[Function]}
        prefixCls="za-calendar"
        value={
          Array [
            2017-11-03T07:00:00.000Z,
            2017-11-06T07:00:00.000Z,
          ]
        }
      >
        <section
          className="za-calendar__month"
          key="2018-4"
          title="2018年5月"
        >
          <ul>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end firstday-2"
              key="2018-4-1"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                1
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-2"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                2
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-3"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                3
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-4"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                4
              </div>
            </li>
            <li
              className="za-calendar__day d6 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-5"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                5
              </div>
            </li>
            <li
              className="za-calendar__day d7 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-6"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                6
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-7"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                7
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-8"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                8
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-9"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                9
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-10"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                10
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-11"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                11
              </div>
            </li>
            <li
              className="za-calendar__day d6 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-12"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                12
              </div>
            </li>
            <li
              className="za-calendar__day d7 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-13"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                13
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-14"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                14
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-15"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                15
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-16"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                16
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-17"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                17
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-18"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                18
              </div>
            </li>
            <li
              className="za-calendar__day d6 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-19"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                19
              </div>
            </li>
            <li
              className="za-calendar__day d7 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-20"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                20
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-21"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                21
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-22"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                22
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-23"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                23
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-24"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                24
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-25"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                25
              </div>
            </li>
            <li
              className="za-calendar__day d6 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-26"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                26
              </div>
            </li>
            <li
              className="za-calendar__day d7 za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-27"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                27
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-28"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                28
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-29"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                29
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-30"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                30
              </div>
            </li>
            <li
              className="za-calendar__day za-calendar__day--selected za-calendar__day--range range-start range-end"
              key="2018-4-31"
              onClick={[Function]}
            >
              <div
                className="za-calendar__day__content"
              >
                31
              </div>
            </li>
          </ul>
        </section>
      </CalendarMonthView>
    </section>
  </div>
</CalendarView>
`;
